model.py

```diff
@@ -6248,10 +6246,35 @@ def clean_media_type(self):
         """
         return self._clean_media_type(self.media_type)
 
+    @property
+    def url_extension(self):
+        """The file extension in this representation's original url."""
+
+        # Known extensions can be followed by a version number (.epub3)
+        # or an additional extension (.epub.noimages)
+        known_extensions = "|".join(self.FILE_EXTENSIONS.values())
+        known_extension_re = re.compile("\.(%s)\d?\.?[\w\d]*$" % known_extensions, re.I)
+
+        known_match = known_extension_re.search(self.url)
+
+        if known_match:
+            return known_match.group()
+
+        else:
+            any_extension_re = re.compile("\.[\w\d]*$", re.I)
+        
+            any_match = any_extension_re.search(self.url)
+
+            if any_match:
+                return any_match.group()
+        return None
```

Comment:
If it's a matter of stripping them, you could use `urlparse.urlparse` on `self.url` and run the regular expression against the `.path` attribute instead of the full URL. To derive useful information out of the query parameter we'd have to write site-specific code.